Fix feedreader tests broken by Python 3.14.3 asyncio changes

### Description

The following `feedreader` tests are flaky/failing due to the Python 3.14.3 asyncio executor scheduling change (see home-assistant/core#162263 for background).

CPython PR python/cpython#142358 changed how `run_in_executor` delivers results back to the event loop. Results are now set directly on the asyncio future instead of going through `call_soon_threadsafe`, which means `await` returns immediately but other `call_soon_threadsafe` callbacks may not have run yet.

#### Affected tests

**`tests/components/feedreader/test_init.py::test_storage_data_writing` (failed 7/10)**

```
tests/components/feedreader/test_init.py:93: in test_storage_data_writing
    assert hass_storage[DOMAIN]["data"] == storage_data
           ^^^^^^^^^^^^^^^^^^^^
E   KeyError: 'feedreader'
```

#### What was done

These tests have been temporarily marked with `@pytest.mark.xfail` in PR #169067.

#### What needs to happen

The codeowner should investigate and fix the underlying timing issue in these tests. Common fixes include:
- Adding `await hass.async_block_till_done()` after service calls
- Converting sync entity methods to async
- Restructuring mocks to avoid unintended executor code paths

### Additional context

- Related PR: home-assistant/core#162263
- Disable PR: #169067

--- a/tests/components/feedreader/test_init.py
+++ b/tests/components/feedreader/test_init.py
@@ -85,6 +85,7 @@ async def test_storage_data_writing(
         assert await async_setup_config_entry(
             hass, VALID_CONFIG_DEFAULT, return_value=feed_one_event
         )
+        await hass.async_block_till_done()
 
     # one new event
     assert len(events) == 1

